batch/batch/worker/worker.py

```diff
@@ -2606,6 +2613,8 @@ async def post_job_complete_1(self, job, full_status):
             delay_secs = min(delay_secs * 2, 2 * 60.0)
 
     async def post_job_complete(self, job, full_status):
+        if job.mjs_fut is not None:
```

Comment:
Should we cancel the future instead? I'm worried about the case where the driver is overloaded and this is just retrying transient errors. Can you remind me why this change is correct?